Application/Frontend/components/FriendsColumn/AddFriendModal.tsx

```diff
@@ -1,88 +1,77 @@
-   * request to the server with the sender's ID and the friend's username. Upon a successful request,
-   * it resets the input field for further requests and triggers a re-fetch of the friend list by updating
-   * the `lastFetched` state. If the request fails or if no username is entered, it sets an appropriate
-   * error message.
-   */
   const handleAddFriendClick = async () => {
-    if (friendUsername) {
+    if (friendUsername.trim()) {
       try {
         const response = await fetch('/api/add-friend', {
           method: 'POST',
           headers: {
             'Content-Type': 'application/json',
           },
-          body: JSON.stringify({ senderID: senderID, friendUsername: friendUsername }),
+          body: JSON.stringify({ senderID, friendUsername }),
         });
 
         setSearchResult(response.status);
-    
+
         if (response.ok) {
-          const data = await response.json();
-          setFriendUsername(''); // Clear the input field to allow for further friend requests
+          setFriendUsername('');
+          // Use Mantine's showNotification for success message
+          showNotification({
+            title: 'Success',
+            message: `Friend request sent to ${friendUsername}!`,
+            color: 'green',
+          });
         } else {
           console.error('Failed to add friend');
+          // Use Mantine's showNotification for error message
+          showNotification({
+            title: 'Error',
+            message: 'Failed to send friend request.',
+            color: 'red',
+          });
         }
       } catch (error) {
         console.error('Error adding friend:', error);
+        // Use Mantine's showNotification for network or server errors
+        showNotification({
+          title: 'Error',
+          message: 'An error occurred while sending the friend request.',
+          color: 'red',
+        });
       }
     } else {
       setErrorMessage('Please enter a username to add a friend.');
     }
   };
 
-    /**
-   * Reacts to changes in the `searchResult` state, which reflects the HTTP status code of the friend
-   * request operation. It sets an error message based on the result (e.g., if the username does not exist).
-   * Upon a successful friend addition, it updates the `lastFetched` timestamp to trigger a re-fetch
-   * of the friend list and closes the modal.
-   */
   useEffect(() => {
-    // This useEffect will only run when searchResult changes
-    switch(searchResult) {
+    switch (searchResult) {
       case 404:
         setErrorMessage('This username does not exist.');
         break;
       default:
-        setErrorMessage('');
-        setLastFetched(Date.now()); // Update the lastFetched timestamp to trigger a refetch of the friend list
-        close();
+        if (searchResult !== null) {
+          setErrorMessage('');
+          setLastFetched(Date.now());
```

Comment:
Excellent use of our custom context that facilitates real-time updates. Great work in using this!